### PR TITLE
orca-slicer: init at 1.8.1

### PR DESCRIPTION
**Pull request by pinpox**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pinpox** [comment]
https://github.com/NixOS/nixpkgs/pull/275638

**Comment by Wulfsta** [comment]
This should not be copied and pasted; look at the way `super-slicer` is packaged. Additionally, you might want to offer @ovlach position as a maintainer as well, since they have been packaging this for months while awaiting `bambu-studio` to land in nixpkgs.

**Comment by pinpox** [comment]
> This should not be copied and pasted; look at the way `super-slicer` is packaged. Additionally, you might want to offer @ovlach position as a maintainer as well, since they have been packaging this for months while awaiting `bambu-studio` to land in nixpkgs.

I just tried do understand what is happening in the `super-slicer` package. Should I be overriding `prusa-slicer` aswell for orca? Not sure My nix skills are up to this PR then, unless anyone can provide further instructions. @ovlach you are of course more than welcome to take the PR over or replace it, just wanted to step in and help getting this into nixpkgs since I'm using it.

**Comment by cab404** [comment]
Any progress with re-doing it with overrides?

**Comment by pinpox** [comment]
> Any progress with re-doing it with overrides?

I'm sorry but I haven't been able to adapte the prusa-slicer package as I'm confused by all thoser versions and patches that get applied. If anyone wants to pick this up, let me know if I can help test otherwise I'll continue using the flake or the less elegant package submitted here

**Comment by Wulfsta** [comment]
> > Any progress with re-doing it with overrides?
> 
> I'm sorry but I haven't been able to adapt the prusa-slicer package as I'm confused by all those versions and patches that get applied. If anyone wants to pick this up, let me know if I can help test otherwise I'll continue using the flake or the less elegant package submitted here
> 
> This was my latest try that build, but the override is not applied correctly: https://gist.github.com/pinpox/75699f3c3468c4171ba4b244912fd1c9

You don’t need to use `prusa-slicer`, just reference it and its relationship to `super-slicer` as the approach you should take to packaging this. This should be a file in the same directory as `bambu-studio` that simply overrides `bambu-studio`.

**Comment by pinpox** [comment]
@Wulfsta I think I got it. Added a file next to `bambu-studio` overriding the changed stuff, it is much simpler than the one from `prusa-slicer` as there is only one version and no patches, but I hope I understood correctly. Let me know if this is the approach you had in mind or if I missed something.

**Comment by Wulfsta** [comment]
> @Wulfsta I think I got it. Added a file next to `bambu-studio` overriding the changed stuff, it is much simpler than the one from `prusa-slicer` as there is only one version and no patches, but I hope I understood correctly. Let me know if this is the approach you had in mind or if I missed something.

Looks good to me! The only issue now is figuring out why borg isn’t happy.

**Comment by pinpox** [comment]
Hm, I don't really understand the error nor can I reproduce it. It build fine on my machine (though it takes for everi on a notebook).

One thing I noticed is, the generated `result` when building with `nix-build -A orca-slicer` points to ` result -> /nix/store/jq6mrspa61ksb6b2g9v7jz46qlixrv36-bambu-studio-bar-01.08.01.57` not sure if that is a problem but it has still `bambu-slicer` in it's name

**Comment by Wulfsta** [comment]
> Hm, I don't really understand the error nor can I reproduce it. It build fine on my machine (though it takes for ever on a notebook).
> 
> One thing I noticed is, the generated `result` when building with `nix-build -A orca-slicer` points to ` result -> /nix/store/jq6mrspa61ksb6b2g9v7jz46qlixrv36-bambu-studio-bar-01.08.01.57` not sure if that is a problem but it has still `bambu-slicer` in it's name

I wont be by a machine to have a closer look for a few weeks, perhaps someone else might be able to help (try asking on matrix if nobody else checks in here)? One other note is that you should add yourself as a maintainer. 🙂

File: pkgs/applications/misc/bambu-studio/orca-slicer.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ lib, fetchFromGitHub, makeDesktopItem, bambu-studio }:
+let
+
+
+  orca-slicer = bambu-studio.overrideAttrs (finalAttrs: previousAttrs: rec {
+
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/bambu-studio/orca-slicer.nix:6]:
```suggestion
  orca-slicer = bambu-studio.overrideAttrs (finalAttrs: previousAttrs: {
```

File: pkgs/applications/misc/bambu-studio/orca-slicer.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    version = "1.8.1";
+    pname = "orca-slicer";
+
+    src = fetchFromGitHub {
+      owner = "SoftFever";
+      repo = "OrcaSlicer";
+      rev = "v${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/bambu-studio/orca-slicer.nix:13]:
```suggestion
      rev = "v${finalAttrs.version}";
```

**Comment by pinpox** [comment]
@Wulfsta Borg is happy! Fix based on recommendations from the IRC channel, had to add platforms to both packages.

**Comment by Wulfsta** [comment]
I think this should be building on Darwin too (at least from memory and a quick look at what was building on the original PR), can you add that?

**Comment by pinpox** [comment]
> I think this should be building on Darwin too

Nope, it does not. Quoting @lilyinstarlight from over on matrix: 
> nixpkgs bambu studio does not eval on darwin and definitely does not build

I don't have any darwin machines to investigate this furthe and would suggest that if someone on darwin is interested in running this, it could be a separate PR with a fix.

**Comment by Wulfsta** [comment]
> > I think this should be building on Darwin too
> 
> Nope, it does not. Quoting @lilyinstarlight from over on matrix:
> 
> > nixpkgs bambu studio does not eval on darwin and definitely does not build
> 
> I don't have any darwin machines to investigate this furthe and would suggest that if someone on darwin is interested in running this, it could be a separate PR with a fix.

Oh awesome, thanks for the correction. Looks good then.

**Review by Wulfsta — APPROVED** [review]

**Comment by mabl** [comment]
Maybe as a heads-up (but it may also be me that does something wrong):

I'm not yet using a Flask system, so I am occasionally using unstable packages via `nix-shell -I nixpkgs=~/.nix-defexpr/channels/nixos-unstable/ -p orca-slicer`.  That has always worked, but somehow fails for this package:

```
$ orca-slicer 
[2024-01-03 19:51:55.870334] [0x00007f940ba3e100] [trace]   Initializing StaticPrintConfigs
add font of HarmonyOS_Sans_SC_Bold returns 1
add font of HarmonyOS_Sans_SC_Regular returns 1
add font of NanumGothic-Regular returns 1
add font of NanumGothic-Bold returns 1
loading network module at /home/mabl/.config/OrcaSlicer/plugins/libbambu_networking.so
error, dlerror is /home/mabl/.config/OrcaSlicer/plugins/libbambu_networking.so: cannot open shared object file: No such file or directory
after dlopen, network_module is (nil)

(orca-slicer:162651): Gtk-CRITICAL **: 19:51:56.369: gtk_cell_layout_get_cells: assertion 'GTK_IS_CELL_LAYOUT (cell_layout)' failed

(orca-slicer:162651): Gtk-CRITICAL **: 19:51:56.369: gtk_cell_layout_get_cells: assertion 'GTK_IS_CELL_LAYOUT (cell_layout)' failed

(orca-slicer:162651): Gtk-CRITICAL **: 19:51:56.370: gtk_cell_layout_get_cells: assertion 'GTK_IS_CELL_LAYOUT (cell_layout)' failed

(orca-slicer:162651): Gtk-CRITICAL **: 19:51:56.371: gtk_cell_layout_get_cells: assertion 'GTK_IS_CELL_LAYOUT (cell_layout)' failed
```

A window is shown, but has no content whatsoever. I am speculating, that it is not using the correct gtk library. At least, it's not picked up by `ldd`:

```
ldd /nix/store/cw730d2n89cmkl9ak812i7568zxxw670-orca-slicer-1.8.1/bin/orca-slicer
        linux-vdso.so.1 (0x00007ffddfdec000)
        libudev.so.1 => /nix/store/iidxwcyp8pqhrq3iji17shs4m6gin0kv-systemd-254.6/lib/libudev.so.1 (0x00007fdb05c0b000)
        libc.so.6 => /nix/store/9y8pmvk8gdwwznmkzxa6pwyah52xy3nk-glibc-2.38-27/lib/libc.so.6 (0x00007fdb05a23000)
        libcap.so.2 => /nix/store/lmj8lfkd8h448grf1nf92lsm5gm2fnf6-libcap-2.69-lib/lib/libcap.so.2 (0x00007fdb05a17000)
        /nix/store/9y8pmvk8gdwwznmkzxa6pwyah52xy3nk-glibc-2.38-27/lib/ld-linux-x86-64.so.2 => /nix/store/9y8pmvk8gdwwznmkzxa6pwyah52xy3nk-glibc-2.38-27/lib64/ld-linux-x86-64.so.2 (0x00007fdb05c50000)
```

This may just be a fluke or may not be expected to work - but I thought it may be worth a warning at least.

**Pull request merged**